02-library/cs506/sim.py

```
@@ -1,13 +1,55 @@
-def euclidean_dist(x, y):
-    raise NotImplementedError()
 
+def euclidean_dist(x, y):
+    #raise NotImplementedError()
+    if (len(x) != len(y)):
+        return "wrong dimensions"
+    vectorSize = len(x)
+    sum = 0
+    for ind in range(vectorSize):
+        sum += (x[ind] - y[ind])**2
+    return sum**(0.5)
+    
 def manhattan_dist(x, y):
-    raise NotImplementedError()
-
+    #raise NotImplementedError()
+    if (len(x) != len(y)):
+        return "wrong dimensions"
+    vectorSize = len(x)
+    sum = 0
+    for ind in range(vectorSize):
+        sum += abs(x[ind] - y[ind])
+    return sum
 def jaccard_dist(x, y):
-    raise NotImplementedError()
-
+    #raise NotImplementedError()
+    if (x == y):
+        return 0
+    a = set(x)
+    b = set(y)
+    
+    #1 - jaccard index
+    intersection_card = len(a.intersection(b))
+    union_card = len(a.union(b))
+    return 1 - intersection_card/union_card
```

can `union_card` be 0? If so, can you raise an error when you encounter that input?